docs/contribute/design_axioms.md

```diff
@@ -0,0 +1,35 @@
+---
+layout: page
+title: Design Axioms
+permalink: /contributing/design-axioms
+---
+
+## 💚 100% Kotlin
+
+Kotlin is the language of choice for components we develop and is the recommended language for application projects integrating components. We make use of Kotlin idioms and official Kotlin extension libraries (e.g. Coroutines).
+
+Java interoperability is not a goal of this project. We may add annotations for Java interoperability (e.g. `@JvmStatic`) at the request of a consumer to facilitate integration, but we are not considering replacing Kotlin idioms for the sake of operability.
+
+## 🔄 Third-party dependencies
+
+As providers of a framework we strive to not depend on other third-party libraries as long as not strictly needed (e.g. Android Jetpack libraries).
+
+It is our goal to let our consumers decide what third-party libraries fit their needs, without our components making this choice or introducing an additional, duplicated stack of third-party frameworks.
+
+Additionally by avoiding excessive frameworks we intent to keep the byte size of our components, and essentially the consuming apps, as small as possible.
+
+## 🧰 Simple and applicable
+
+We are building components that directly satisfy the needs of our consumers. We tend to not design components in a vacuum without a defined use case. Components are preferred to be simple and "to the point". Additional functionality is added as needed, never in advance without knowing if it will ever be used (or how).
+
+## 🎨 Customizability
+
+Different consumers have different requirements. If possible we try to provide customization options (e.g. styling) as and if needed by consumers. We try to balance customizability and complexity. Sometimes we prefer alternative component implementations over too complex customization options.
```

Comment:
nit: [...] "If possible, we try to provide customization options (e.g. styling) as needed by the consumers. We aim to balance customizability and complexity. When too complex of customization options are required, we prefer alternative component implementations."